cmd/clusterctl/hack/create-local-repository.py

```diff
@@ -130,6 +131,10 @@ def write_local_repository(provider, version, components_file, components_yaml):
         f.write(components_yaml)
         f.close()
 
+        # Copy CAPI metadata for non-infrastructure providers
+        if not provider.startswith("infrastructure"):
```

Comment:
Why are are excluding `infrastructure` providers? At least for infrastructure docker we should copy the same file, but now all the providers requires metadata.yaml...

What about adding `additionalFiles` (array) to the config for each provider, so we allow to create a local repository with metadata, templates etc. etc. for all the infrastructure providers. If this is ok for you, we should also update developer instructions in https://cluster-api.sigs.k8s.io/clusterctl/developers.html#create-a-clusterctl-settingsjson-file

Do you mind to upgrade the provider version in providers from 'v0.3.8' to 'v0.4.0'?